algebird-core/src/main/scala/com/twitter/algebird/ExpHist.scala

```diff
@@ -0,0 +1,199 @@
+package com.twitter.algebird
+
+/**
+ * Exponential Histogram algorithm from
+ * http://ilpubs.stanford.edu:8090/504/1/2001-34.pdf.
+ *
+ *
+ * Next step - code up the l-canonical representation of numbers,
+ * then do some of the fancier business in the paper!
+ */
+object ExpHist {
+  case class Config(k: Int, windowSize: Long) {
+    // Maximum number of buckets of size 2^i allowed in the repr of
+    // this exponential histogram.
+    def maxBuckets: Int = math.ceil(k / 2.0).toInt + 1
```

Comment:
the paper states that there can't be more than `ceil(k/2) + 2` buckets of size `2^i` for any `i`.